### PR TITLE
[Opset15][Spec] Add and update specifications for EmbeddingBag ops

### PR DESCRIPTION
**Pull request by mmikolajcz**
### Details:
 - *Add specification for EmbeddingBagOffsets-15 and EmbeddingBagPacked-15*
 - *Remove dead links and improve specification of EmbeddingBagOffsetsSum-3 and EmbeddingBagPackedSum-3*

### Tickets:
 - *141862*
 - *141863*

### CODE REVIEW

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-packed-15.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+* **1**: tensor of shape ``[batch, emb_dim1, emb_dim2, ...]`` and of type *T* containing embeddings for each bag.
+
+**Types**
+
+* *T*: any numeric type.
+* *T_IND*: ``int32`` or ``int64``.
+
+**Example**
+
+.. code-block:: cpp
```
Comment (mitruska) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-packed-15.rst:56]:
```suggestion
.. code-block:: xml
   :force:
```

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-offsets-15.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  * **Type**: ``string``
+  * **Default value**: sum
+  * **Required**: *no*
+
+**Inputs**:
+
+* **1**: ``emb_table`` tensor containing the embedding lookup table of the module of shape ``[num_emb, emb_dim1, emb_dim2, ...]`` and  of type *T*. **Required.**
+* **2**: ``indices`` tensor of shape ``[num_indices]`` and of type *T_IND*. **Required.**
+* **3**: ``offsets`` tensor of shape ``[batch]`` and of type *T_IND* containing the starting index positions of each "bag" in ``indices``. Maximum value of offsets cannot be greater than length of ``indices``. **Required.**
+* **4**: ``default_index`` scalar of type *T_IND* containing default index in embedding table to fill empty "bags". If set to ``-1`` or not provided, empty "bags" are filled with zeros. Reverse indexing using negative values is not supported. **Optional.**
```
Comment (mitruska) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-offsets-15.rst:44]: Usage of `-1` is not an ideal solution, but as we discussed it's aligned with current behavior of some plugins and it resolves the empty bags issues. Although the main reason of adding v15 version was enablement of the "mean" mode, great to see the clarification of the `default_index` behavior as well.

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-packed-15.rst (hunk 1 above)
Comment (mmikolajcz) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-packed-15.rst]: Done

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-offsets-15.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+**Category**: *Sparse*
+
+**Short description**: Computes sums or means of "bags" of embeddings, without instantiating the intermediate embeddings.
+
+**Detailed description**:
+
+Operation EmbeddingBagOffsets is an implementation of ``torch.nn.EmbeddingBag`` with indices and offsets inputs being 1D tensors.
+
+For each index in ``indices`` this operator gathers values from ``emb_table`` embedding table. Indices that are in range of same bag based on ``offset`` input values are then reduced according to ``reduction`` attribute.
```
[the remark this answers is not in the document]
Comment (mmikolajcz) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-offsets-15.rst]: Applied + added NumPy snippet to explain whole op in more details

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-offsets-15.rst [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+* *T*: any numeric type.
+* *T_IND*: ``int32`` or ``int64``.
+
+**Example**
+
+.. code-block:: xml
+
+   <layer ... type="EmbeddingBagOffsets" ... >
+       <input>
```
Comment (mitruska) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-offsets-15.rst:103]: Usually every example has a title, it should be added to emphasise what is presented.
Please take a look at any other spec for reference, for instance: 
https://github.com/openvinotoolkit/openvino/blob/55e3bee5fd20ab726595bba6c4040c16749fe6b5/docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sequence/scaled-dot-product-attention.rst?plain=1#L95-L100

Also would be great to add an example of EmbeddingBagOffsets with new reduction attribute and "mean" value.

Comment (mmikolajcz) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/sparse/embedding-bag-offsets-15.rst:103]: Done, + adjusted per_sample_weights to better show difference between sum and mean

**Review by mitruska — APPROVED** [review]

**Pull request merged**